firestore/src/main/java/com/example/firestore/snippets/QueryDataSnippets.java

```diff
@@ -328,4 +335,51 @@ void createMultipleCursorConditionsQuery() {
         .startAt("Springfield", "Missouri");
     // [END fs_multiple_cursor_conditions]
   }
+
+  /**
+   * Create a query using a snapshot as a start point.
+   *
+   * @return query
+   */
+  Query createStartAtSnapshotQueryCursor()
+      throws InterruptedException, ExecutionException, TimeoutException {
+    // [START fs_document_snapshot_cursor]
+    // Fetch the Snapshot
+    ApiFuture<DocumentSnapshot> future = db.collection("cities").document("SF").get();
+    DocumentSnapshot snapshot = future.get(30, TimeUnit.SECONDS);
+
+    // Construct the query
+    Query query = db.collection("cities")
+        .orderBy("population")
+        .startAt(snapshot);
+    // [END fs_document_snapshot_cursor]
+    return query;
+  }
+
+  /**
+   * Example of a paginated query.
+   */
+  void paginateCursor() throws InterruptedException, ExecutionException, TimeoutException {
+    // [START fs_paginate_cursor]
+    // Construct query for first 25 cities, ordered by population.
+    CollectionReference cities = db.collection("cities");
+    ApiFuture<QuerySnapshot> firstPage = cities
+        .orderBy("population")
+        .limit(25)
+        .get();
+
+    // Wait for results.
+    List<QueryDocumentSnapshot> docs = firstPage.get(30, TimeUnit.SECONDS).getDocuments();
```

Comment:
This is performing another `get`? I'm a little confused by snippet given it calls `get`: one here and another on the previous line.

Reply:
Thanks!